0012-five-bells-ledger-api/0012-five-bells-ledger-api.md

```diff
@@ -0,0 +1,1290 @@
+
+#### Response Format
+
+A successful result uses the HTTP response code **200 OK** and contains a JSON body with the [Transfer resource][] as saved.
+
+#### Example
+
+Request:
+
+```json
+PUT /transfers/3a2a1d9e-8640-4d2d-b06c-84f2cd613204
+Content-Type: application/json
+
+{
+  "id": "3a2a1d9e-8640-4d2d-b06c-84f2cd613204",
+  "debits": [{
+    "account": "https://red.ilpdemo.org/ledger/accounts/alice",
+    "amount": "50",
+    "authorized": true
+  }],
+  "credits": [{
+    "account": "https://red.ilpdemo.org/ledger/accounts/bob",
+    "amount": "50"
+  }],
+  "execution_condition": "cc:0:3:8ZdpKBDUV-KX_OnFZTsCWB_5mlCFI3DynX5f5H2dN-Y:2",
+  "expires_at": "2015-06-16T00:00:01.000Z"
+}
+```
+
+Response:
+
+```json
+HTTP/1.1 200 OK
+
+
+```
+
+#### Errors
+
+- [InsufficientFundsError][] - A debited account would go below its `minimum_allowed_balance` if this transfer were executed
+- [UnprocessableEntityError][] - The request is formatted properly but contains an otherwise-unspecified semantic problem.
+- [AlreadyExistsError][] - The `id` supplied is already used by another transfer.
+- [InvalidUriParameterError][] - One of the [URL][] or [UUID][] parameters was not formatted properly.
+- [InvalidBodyError][] - The request body was not properly-formatted JSON, did not match the Content-Type provided, or did not match the schema for the requested resource type.
+- [UnsupportedCryptoConditionError][] - The transfer included an `execution_condition` whose feature bitstring requires functionality not implemented by this ledger.
+
+
+### Submit Fulfillment
+[Submit Fulfillment]: #submit-fulfillment
+
+Execute or cancel a transfer by submitting a [Crypto-Condition Fulfillment][].
+
+To execute a transfer, the transfer MUST begin in the `prepared` state and the submitted fulfillment must satisfy the [Crypto-Condition][] in the transfer's `execution_condition` field. Doing so transitions the transaction to `executed`.
+
+To cancel a transfer, the submitted fulfillment must satisfy the [Crypto-Condition][] in the transfer's `cancellation_condition` field. Doing so transitions the transaction to `rejected`.
+
+**Authorization:** The owner of an account MUST be able to submit the fulfillment for a transfer that credits the account. A ledger MAY allow other account owners or unauthorized clients to submit a fulfillment.
+
+#### Request Format
+
+
+```
+PUT /transfers/{id}/fulfillment
+Content-Type: text/plain
+```
+
+**Caution:** This method requires the request to specify the header `Content-Type: text/plain`
+
+**[Metadata URL Name][]:** `transfer_fulfillment`.
+
+##### Body Parameters
+
+The message body should be a [Crypto-Condition Fulfillment][] in string format.
+
+#### Response Format
+
+A successful result uses the HTTP response code **200 OK**. The message body of the response is the submitted fulfillment as plain text. A ledger MUST return a successful response if and only if the transfer was executed or canceled as a result of this request.
```

Comment:
@sentientwaffle Can you tell me whether the "Submit Fulfillment" response returns 201 the first time a fulfillment causes a transfer to be executed or canceled?